### PR TITLE
Admin Menu: Keep Plugins submenus when they link to WP Admin

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/jetpack/issues/18831

#### Changes proposed in this Pull Request:
- Add back the submenus of the Plugins menu when they link to WP Admin (only Atomic sites).
- Force the Plugins menu to link to Calypso on Simple and Jetpack sites (to mimic the behavior of the existing/old navigation).

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_Simple_
* Go to https://wordpress.com/me/account and disable the "Replace all dashboard pages with WP Admin equivalents when possible." setting.
* Apply D57572-code to your WP.com sandbox.
* Sandbox the API and a Simple site.
* Go to https://wordpress.com and switch to that site.
* Make sure that the Plugins menu links to Calypso.
* Enable now the "Replace all dashboard pages with WP Admin equivalents when possible." setting.
* Make sure that Plugins still links to Calypso.

_Jetpack_
* Go to https://wordpress.com/me/account and disable the "Replace all dashboard pages with WP Admin equivalents when possible." setting.
* Spin up a Jetpack site running the branch of this PR.
* Go to https://wordpress.com and switch to that site.
* Make sure that the Plugins menu links to Calypso.
* Enable now the "Replace all dashboard pages with WP Admin equivalents when possible." setting.
* Make sure that Plugins still links to Calypso.

_Atomic_
* Go to https://wordpress.com/me/account and disable the "Replace all dashboard pages with WP Admin equivalents when possible." setting.
* Install Jetpack Beta on a WoA site up and activate the branch of this PR.
* Go to https://wordpress.com and switch to that site.
* Make sure that the Plugins menu contains 3 submenus: Installed Plugins, Add New, and Plugin Editor.
* Make sure they all link to WP Admin.
* Enable now the "Replace all dashboard pages with WP Admin equivalents when possible." setting
* Make sure that the Plugins submenus still point to WP Admin.

#### Proposed changelog entry for your changes:
N/A.

### CODE REVIEW

File: projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php [hunk 1]
```diff
@@ -180,4 +180,18 @@ public function test_add_appearance_menu() {
...
+
+	/**
+	 * Tests add_plugins_menu
+	 *
+	 * @covers ::add_plugins_menu
+	 */
+	public function test_add_plugins_menu() {
+		global $menu;
+
+		static::$admin_menu->add_plugins_menu( true );
```
Comment (tjcafferkey) [inline projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php:192]: Do we need an assertion to check that when `add_plugins_menu` is given `false`?

File: projects/plugins/jetpack/tests/php/modules/masterbar/test-class-wpcom-admin-menu.php [hunk 2]
```diff
@@ -345,4 +345,18 @@ public function test_add_gutenberg_menus() {
...
+
+	/**
+	 * Tests add_plugins_menu
+	 *
+	 * @covers ::add_plugins_menu
+	 */
+	public function test_add_plugins_menu() {
+		global $menu;
+
+		static::$admin_menu->add_plugins_menu( true );
```
Comment (tjcafferkey) [inline projects/plugins/jetpack/tests/php/modules/masterbar/test-class-wpcom-admin-menu.php:357]: Do we need an assertion to check that when `add_plugins_menu` is given `false`?

File: projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php (hunk 1 above)
Comment (mmtr) [reply projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php:192]: That's already covered by the default tests done in the `Test_Admin_Menu` class. Here we're only testing the implementation differences that are done in the menu for Jetpack sites.

File: projects/plugins/jetpack/tests/php/modules/masterbar/test-class-wpcom-admin-menu.php (hunk 2 above)
Comment (mmtr) [reply projects/plugins/jetpack/tests/php/modules/masterbar/test-class-wpcom-admin-menu.php:357]: That's already covered by the default tests done in the `Test_Admin_Menu` class. Here we're only testing the implementation differences that are done in the menu for Simple sites.

**Review by tjcafferkey — APPROVED** [review]
Thank you for the thorough testing instructions, always appreciated. This tests well for me as described and the code LGTM 🚢

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.5` in 3fcd59632c3c10980779ab0fb893b2ea94f8083f

**Pull request merged**